Library/Formula/fstar.rb

```diff
@@ -0,0 +1,94 @@
+class Fstar < Formula
+  desc "F*, an ML-like language with a type system for program verification"
+  homepage "http://fstar.gforge.inria.fr/"
+  url "https://github.com/FStarLang/FStar.git",
+    :tag => "v0.9.0",
+    :revision => "f24457213f4007302873f6d42eb2cf8094fa45d0"
+  head "https://github.com/FStarLang/FStar.git"
+
+  depends_on "mono"   => :build
+  depends_on "fsharp" => :build
+  depends_on "opam"   => :build
+  depends_on "ocaml"  => :recommended
+  depends_on "homebrew/science/z3"
```

Comment:
Core formulae can’t have hard dependencies in taps. We need to have this optional or move this formula in the `homebrew/science` tap.